File: lib/src/model/canned_access_control_list.dart
enum CannedAccessControlList {
  Private,

  PublicRead,

  PublicReadWrite,

  Default,
}

extension on CannedAccessControlList {
  String get customName {
    switch (this) {
      case CannedAccessControlList.Private:
        return 'private';
      case CannedAccessControlList.PublicRead:
        return 'public-read';
      case CannedAccessControlList.PublicReadWrite:
        return 'public-read-write';
      case CannedAccessControlList.Default:
      default:
        return 'default';
    }
  }
}

CannedAccessControlList? parseACL(String aclStr) {
  CannedAccessControlList? currentAcl;
  for (CannedAccessControlList acl in CannedAccessControlList.values) {
    if (acl.customName == aclStr) {
      currentAcl = acl;
      break;
    }
  }
  return currentAcl;
}

File: lib/src/model/CannedAccessControlList.java
package com.alibaba.sdk.android.oss.model;

/**
 * bucket ACL enum definition
 * Created by LK on 15/12/17.
 */
 enum CannedAccessControlList {

    (""),

    Read("-read"),

    ReadWrite("-read-write"),

    Default("default");

     String ACLString;

    CannedAccessControlList(String acl) {
        this.ACLString = acl;
    }

     static CannedAccessControlList parseACL(String aclStr) {
        CannedAccessControlList currentAcl = null;
        for (CannedAccessControlList acl : CannedAccessControlList.values()) {
            if (acl.toString().equals(aclStr)) {
                currentAcl = acl;
                break;
            }
        }
        return currentAcl;
    }

    @Override
     String toString() {
        return this.ACLString;
    }
}
